### DISCUSSION TITLE
Apps should have runtipi label

### DISCUSSION THREAD

**Discussion by ahmedrowaihi** [runtipi/runtipi-appstore#3919] [Ideas]
all apps compose.yml in store should have label ```runtipi.managed=true```
this way host could easily distinguish between apps running managed by runtipi or it's by other processes

it also make uninstalls/pruning easier with docker filter ```docker container prune -f --filter "label=runtipi.managed=true"```

**Comment by JigSawFr** [comment]
Totally agree, @meienberger should be easy to add. Think that in the main docker compose that all apps inherit no ?

**Reply by ahmedrowaihi** [reply]
I don't see a docker compose that apps inherit from, seems each app has its own docker compose yml

**Reply by ahmedrowaihi** [reply]
but no problem, if you guys agreed on this, I wll update them all in a PR, just confirm

**Reply by nicotsx** [reply]
I like this idea as well

**Reply by nicotsx** [reply]
There is docker-compose.common.yml but I don’t think we can add labels to a service with this kind of file composition

**Reply by ahmedrowaihi** [reply]
@meienberger no worries, I will do it

**Reply by ahmedrowaihi** [reply]
#3931 started

**Reply by ahmedrowaihi** [reply]
if you wonder how label was added,
```javascript
import yaml from 'js-yaml';
import fs from 'node:fs';
import fg from 'fast-glob';

fg.globSync(['./apps/**/docker-compose.yml']).forEach((file) => {
  const content = fs.readFileSync(file, 'utf8');

  const compose = yaml.load(content);
  const services = compose.services;

  for (const service in services) {
    services[service].labels = services[service].labels || {};
    services[service].labels['runtipi.managed'] = true;
  }

  const newContent = yaml.dump(compose);

  fs.writeFileSync(file, newContent, 'utf8');
});
```

after that, I went through each file and staged only the label addition
